modules/png-extraction/ImageExtractor.py

```diff
@@ -273,142 +288,226 @@ def fix_mismatch(with_VRs=['PN', 'DS', 'IS']):
-		
-#load every metadata file using only valid columns 
-meta_list = list()
-for meta in metas:
-    m = pd.read_csv(meta,dtype='str',usecols=loadable_names)
-    meta_list.append(m)
-merged_meta = pd.concat(meta_list,ignore_index=True)
-merged_meta.to_csv('{}/metadata.csv'.format(output_directory),index=False)
-#getting a single mapping file
-logging.info('Generatign final mapping file')
-mappings = glob.glob("{}/maps/*.csv".format(output_directory))
-map_list = list()
-for mapping in mappings:
-    map_list.append(pd.read_csv(mapping,dtype='str'))
-merged_maps = pd.concat(map_list,ignore_index=True)
-if print_only_common_headers:
-    mask_common_fields = merged_maps.isnull().mean() < 0.1
-    common_fields = set(np.asarray(merged_maps.columns)[mask_common_fields])
-    merged_maps = merged_maps[common_fields]
-merged_maps.to_csv('{}/mapping.csv'.format(output_directory),index=False)
-
-
-if send_email:
-    subprocess.call('echo "Niffler has successfully completed the png conversion" | mail -s "The image conversion has been complete" {0}'.format(email), shell=True)
-# Record the total run-time
-logging.info('Total run time: %s %s', time.time() - t_start, ' seconds!')
+                all_headers[e] = 1
+
+    loadable_names = list()
+    for k in col_names.keys():
+        if k in all_headers and all_headers[k] >= no_splits:  # no_splits == number of batches used 
+            loadable_names.append(k) # use header only if it's present in every metadata file
+            
+    #load every metadata file using only valid columns 
+    meta_list = list()
+    for meta in metas:
+        m = pd.read_csv(meta,dtype='str',usecols=loadable_names)
+        meta_list.append(m)
+    merged_meta = pd.concat(meta_list,ignore_index=True)
+    merged_meta.to_csv('{}/metadata.csv'.format(output_directory),index=False)
+    #getting a single mapping file
+    logging.info('Generatign final mapping file')
+    mappings = glob.glob("{}/maps/*.csv".format(output_directory))
+    map_list = list()
+    for mapping in mappings:
+        map_list.append(pd.read_csv(mapping,dtype='str'))
+    merged_maps = pd.concat(map_list,ignore_index=True)
+    if (print_only_common_headers == 'True' or print_only_common_headers == 'true'):
+        mask_common_fields = merged_maps.isnull().mean() < 0.1
+        common_fields = set(np.asarray(merged_maps.columns)[mask_common_fields])
+        merged_maps = merged_maps[common_fields]
+    merged_maps.to_csv('{}/mapping.csv'.format(output_directory),index=False)
+
+    if (send_email == 'True' or send_email == 'true'):
+       subprocess.call('echo "Niffler has successfully completed the png conversion" | mail -s "The image conversion has been complete" {0}'.format(email), shell=True)
+    # Record the total run-time
+    logging.info('Total run time: %s %s', time.time() - t_start, ' seconds!')
+    logs = []
+    logs.append(err)
+    logs.append("DONE")
```

Comment:
What is this "DONE"? 

I think line 450 can be removed.

Reply:
Or did you mean, "the extraction is complete?"

If so, best to give a more descriptive log.
    logs.append("The PNG Extraction is complete")

Reply:
The "DONE" is for frontend status which the user checks.

Reply:
ok, then fine.

Perhaps, a bit more verbose log, like, "The PNG conversion is SUCCESSFUL"?

Reply:
Okay, that would look more appropriate.